presto-docs/src/main/sphinx/installation/spark.rst

```diff
@@ -0,0 +1,44 @@
+=========================
+Executing Presto on Spark
+=========================
+
+When there is a need to execute long running queries, Presto offers
+an option to scale the resource management using Spark, as it doesn'f offer
+one currently by itself. This way Presto can be executed as a stand alone 
+application on Spark.
+
+Steps
+-----
+
+Download the Presto Spark package tarball, :maven_download:`spark-package` 
+and the Presto Spark laucher, :maven_download:`spark-launcher`. Keep both the
+files at, say, *example* directory.
+
+Presto needs``config.properties``. The following is a minimal 
+``config.properties``:
+
+.. code-block:: none
+
+    query.hash-partition-count=10
+    redistribute-writes=false
+ 
+To execute Presto on Spark, first start your Spark cluster, which we will 
+assume have the URL *spark://spark-master:7077*. Keep your 
+time consuming query in a file called, say, *query.sql*.
+
+.. parsed-literal::
+
+    /spark/bin/spark-submit \
+    --master spark://spark-master:7077 \
+    --executor-cores 4 \
```

Comment:
Please add a paragraph explaining this. We need to set `spark.task.cpus` equal to the `--executor-cores` to make sure only a single Presto on Spark task is run in a single Spark executor. We introduced this limitation to avoid duplicating broadcasted hash table for every task. We may optimize this in the future and remove this limitation. Also the `task.concurrency`, `task.max-worker-threads` (and possibly `task.writer-count`) should be adjusted to keep all executor cores busy. For example if `--executor-cores 4` the `task.concurrency`, `task.max-worker-threads` must be set to `4`. You can actually include this properties in the `config.properties` example.